modApiServer/test/org/aion/api/server/ApiUtilTests.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+
+package org.aion.api.server;
+
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class ApiUtilTests {
+
+    private byte vers, retCode;
+
+    private byte errorLength;
+    private byte resultLength;
+
+    private byte[] hash;
+    private byte[] error;
+    private byte[] result;
+
+    public ApiUtilTests() {
+        errorLength = 7;
+        resultLength = 6;
+        vers = RandomUtils.nextBytes(1)[0];
+        retCode = RandomUtils.nextBytes(1)[0];
+        hash = RandomUtils.nextBytes(ApiUtil.HASH_LEN);
+        error = RandomUtils.nextBytes(errorLength);
+        result = RandomUtils.nextBytes(resultLength);
+    }
+
+    @Test
+    public void TestNulls() {
+        System.out.println("run TestNulls.");
+
+        assertNull(ApiUtil.toReturnHeader(vers, retCode, null));
```

Comment:
Could you print out the variables that are randomized, so if this fails in CI server, we can reproduce it?

You might have to do that in every test method because IIRC, I think JUnit reconstructs the test class for each test method

Reply:
done